src/commands/app/deploy.js

```diff
@@ -146,7 +146,11 @@ class Deploy extends BuildCommand {
 
       if (flags['feature-event-hooks']) {
         this.log('feature-event-hooks is enabled, running pre-deploy-event-reg hook')
-        await this.config.runHook('pre-deploy-event-reg', { appConfig: config })
+        const hookResults = await this.config.runHook('pre-deploy-event-reg', { appConfig: config })
+        if (hookResults?.failures?.length > 0) {
+          // output should be "Error : <plugin-name> : <error-message>\n" for each failure
+          this.error(hookResults.failures.map(f => `${f.plugin.name} : ${f.error.message}`).join('\nError: '), { exit: 1 })
```

Comment:
Do we want to extract this into a function in the class? Since it is duplicated thrice

Reply:
its not exactly the same all 3 times
if we need to write it 5 times I'll consider it, or if it expands into more than 1 line.